lib/dal/src/action.rs

```diff
@@ -545,19 +548,57 @@ impl Action {
         Ok(resource)
     }
 
+    /// An Action is dispatchable if all of the following are true:
+    ///   * The action is in the state [`ActionState::Queued`](ActionState)
+    ///   * The graph of values for `DependentValuesUpdate` does *NOT* include
+    ///     *ANY* [`AttributeValue`s](AttributeValue) for the same
+    ///     [`Component`](crate::Component) as the [`Action`].
+    ///
+    /// This method **DOES NOT** check the `DependentValuesUpdate` graph. That
+    /// is done as part of [`Self::eligible_to_dispatch()`]
     pub fn is_eligible_to_dispatch(&self) -> bool {
         // Only Actions in the ActionState::Queued state are dispatchable.
         self.state() == ActionState::Queued
     }
 
+    /// An Action is dispatchable if all of the following are true:
+    ///   * The action is in the state [`ActionState::Queued`](ActionState)
+    ///   * The graph of values for `DependentValuesUpdate` does *NOT* include
+    ///     *ANY* [`AttributeValue`s](AttributeValue) for the same
+    ///     [`Component`](crate::Component) as the [`Action`].
     pub async fn eligible_to_dispatch(ctx: &DalContext) -> ActionResult<Vec<ActionId>> {
         let action_dependency_graph = ActionDependencyGraph::for_workspace(ctx).await?;
         let mut result = Vec::new();
+        let dependent_value_graph = DependentValueGraph::new(
+            ctx,
+            ctx.workspace_snapshot()?
+                .list_dependent_value_value_ids()
+                .await?,
+        )
+        .await?;
+
+        // Find the ComponentIds for all AttributeValues in the full dependency graph for the
+        // queued/running DependentValuesUpdate. We'll want to hold off on dispatching any Actions
+        // that would be operating on the same Component until after the DependentValuesUpdate has
+        // finished working with that Component.
+        let mut dvu_component_ids: HashSet<ComponentId> = HashSet::new();
+        for av_id in &dependent_value_graph.all_value_ids() {
+            dvu_component_ids.insert(AttributeValue::component_id(ctx, *av_id).await?);
+        }
 
         for possible_action_id in action_dependency_graph.independent_actions() {
             let action = Action::get_by_id(ctx, possible_action_id).await?;
 
             if action.is_eligible_to_dispatch() {
+                if let Some(action_component_id) = Action::component_id(ctx, action.id()).await? {
```

Comment:
would we ever want this to get more granular? 

I haven't thought this all of the way through, but if a qualification hasn't finished yet for a component, we could still run the action?

Reply:
If we were to want to do that in the future, we'd want to extend the graph to also contain what kind of thing each AV is for, and record the information as we're building up the graph.

To complicate things further: While we don't do this now, with the way we've structured things, it is possible for the result of a qualification to be an input to another function (it's possible to expose qualification results on an output socket, or even to use them in another Prop). So, while it's possible we might not want to wait on the qualification(s) in that case, we'd still need to build up the full graph, and consider each AttributeValue individually as to whether or not to include it's Component ID in the list of blockers (instead of filtering the initial set of "root" AV IDs).

So, the highlighted check wouldn't change; we'd be populating the set of ComponentIds with more consideration as to what that individual AttributeValue "is".